updater.js

```diff
@@ -0,0 +1,233 @@
+require("dotenv").config();
+const util = require("util");
+const exec = util.promisify(require("child_process").exec);
+const versionNumberRegexp = /([vV])?([0-9]{1,2})\.([0-9]{1,2})(?:\.([0-9]{1,2}))?(?:\.([0-9]{1,2}))?[\n|\s]?/g;
+const wallets = require("./wallets");
+const writeFile = util.promisify(require("fs").writeFile);
+const manifest = require("./package");
+
+class Version {
+  constructor(versionRegexpResult) {
+    this.prefix = versionRegexpResult[1] || "";
+    this.major = parseInt(versionRegexpResult[2]);
+    this.minor = parseInt(versionRegexpResult[3]);
+    this.patch = versionRegexpResult[4] ? parseInt(versionRegexpResult[4]) : null;
+    this.fourth = versionRegexpResult[5] ? parseInt(versionRegexpResult[5]) : null;
+  }
+
+  static fromVersionString(versionString) {
+    if (typeof versionString === "string") {
+      versionNumberRegexp.lastIndex = 0;
+      let regexpResult = versionNumberRegexp.exec(versionString);
+      if (regexpResult) {
+        return new Version(regexpResult);
+      }
+    }
+
+    throw new Error("Can't parse version string : syntax error");
+  }
+
+  toString() {
+    let string = this.prefix + this.major + "." + this.minor;
+    string += this.patch !== null ? "." + this.patch : "";
+    string += this.fourth !== null ? "." + this.fourth : "";
+    return string;
+  }
+}
+
+class Update {
+  constructor(walletIdentifier, from, to) {
+    this.walletIdentifier = walletIdentifier;
+    this.from = from;
+    this.to = to;
+  }
+
+  toString() {
+    return "updated " + this.walletIdentifier + " from " + this.from.toString() + " to " + this.to.toString();
+  }
+}
+
+(async function() {
+  try {
+    console.log("--- Wallet updater launched at " + new Date() + "--");
+    let updates = await checkAllForUpdates();
+
+    if (updates.filter(o => { return o !== null; }).length > 0) {
+      await updateFile(updates);
+      await updatePatchNumber();
+
+      const commitMessage = buildCommitMessage(updates);
+
+      await addDeployKey();
+      await addChanges();
+      await commit(commitMessage);
+      await tag(commitMessage);
+      await push();
+      await publish();
+    } else {
+      console.log("Wallets are up to date.");
+    }
+
+    console.log("--- Wallet updater ended at " + new Date() + "--");
+  } catch (e) {
+    console.error(e);
+  }
+})();
+
+
+async function addDeployKey() {
+  if (!process.env.deploy_key) {
+    throw new Error("Environment variable deploy_key is not set - cannot send modifications to server.");
+  }
+  const { stdout, stderr } = await exec("eval \"$(ssh-agent -s)\" && echo $deploy_key | ssh-add -");
+  console.log(stdout);
+  console.log(stderr);
+}
+
+async function addChanges() {
```

Comment:
You could create a helper to create these methods, and plz use spawn instead of exec, but be careful of the different synthax.